homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1037,196 +952,6 @@ async def async_set_led_brightness(self, brightness: int = 2):
-
-    @property
-    def speed_list(self) -> list:
-        """Get the list of available speeds."""
-        return self._speed_list
-
-    @property
-    def speed(self):
-        """Return the current speed."""
-        if self._state:
-            return AirhumidifierOperationMode(self._state_attrs[ATTR_MODE]).name
-
-        return None
-
-    async def async_set_speed(self, speed: str) -> None:
-        """Set the speed of the fan."""
-        if self.supported_features & SUPPORT_SET_SPEED == 0:
-            return
-
-        _LOGGER.debug("Setting the operation mode to: %s", speed)
-
-        await self._try_command(
-            "Setting operation mode of the miio device failed.",
-            self._device.set_mode,
-            AirhumidifierOperationMode[speed.title()],
-        )
-
-    async def async_set_led_brightness(self, brightness: int = 2):
-        """Set the led brightness."""
-        if self._device_features & FEATURE_SET_LED_BRIGHTNESS == 0:
-            return
-
-        await self._try_command(
-            "Setting the led brightness of the miio device failed.",
-            self._device.set_led_brightness,
-            AirhumidifierLedBrightness(brightness),
-        )
-
-    async def async_set_target_humidity(self, humidity: int = 40):
-        """Set the target humidity."""
-        if self._device_features & FEATURE_SET_TARGET_HUMIDITY == 0:
-            return
-
-        await self._try_command(
-            "Setting the target humidity of the miio device failed.",
-            self._device.set_target_humidity,
-            humidity,
-        )
-
-    async def async_set_dry_on(self):
-        """Turn the dry mode on."""
-        if self._device_features & FEATURE_SET_DRY == 0:
-            return
-
-        await self._try_command(
-            "Turning the dry mode of the miio device off failed.",
-            self._device.set_dry,
-            True,
-        )
-
-    async def async_set_dry_off(self):
-        """Turn the dry mode off."""
-        if self._device_features & FEATURE_SET_DRY == 0:
-            return
-
-        await self._try_command(
-            "Turning the dry mode of the miio device off failed.",
-            self._device.set_dry,
-            False,
-        )
-
-
-class XiaomiAirHumidifierMiot(XiaomiAirHumidifier):
-    """Representation of a Xiaomi Air Humidifier (MiOT protocol)."""
-
-    MODE_MAPPING = {
-        AirhumidifierMiotOperationMode.Low: SPEED_LOW,
-        AirhumidifierMiotOperationMode.Mid: SPEED_MEDIUM,
-        AirhumidifierMiotOperationMode.High: SPEED_HIGH,
-    }
-
-    REVERSE_MODE_MAPPING = {v: k for k, v in MODE_MAPPING.items()}
-
-    @property
-    def speed(self):
-        """Return the current speed."""
-        if self._state:
-            return self.MODE_MAPPING.get(
-                AirhumidifierMiotOperationMode(self._state_attrs[ATTR_MODE])
-            )
-
-        return None
-
-    @property
-    def button_pressed(self):
```

Comment:
Can we expose this as a sensor instead?

Reply:
We could, but what does it add? It does not seem usefull to me.

Reply:
To demonstrate how a `button_pressed` sensor would look see the screenshot.
![afbeelding](https://user-images.githubusercontent.com/7188918/122643913-b6b71380-d112-11eb-9b95-687605f8509c.png)
The `button_pressed` attribute only indicates the last button pressed. If the same button is pressed again, there will be not state change. I don't think this is very useful. Not all humidifier models support the attribute, so that would need additional logic as well. But as you can see adding sensor is easy now.